operators/src/processing/expression/parser.rs

```diff
@@ -285,35 +282,22 @@ impl ExpressionParser {
 
     fn resolve_boolean_expression_rule(&self, pair: Pair<Rule>) -> Result<BooleanExpression> {
         match pair.as_rule() {
-            Rule::identifier => {
-                let identifier = pair.as_str().into();
-                if self.boolean_parameters.contains(&identifier) {
-                    Ok(BooleanExpression::Variable(identifier))
-                } else {
-                    Err(ExpressionError::UnknownBooleanVariable {
-                        variable: identifier.to_string(),
-                    })
-                }
-            }
             Rule::identifier_is_nodata => {
-                // convert `A IS NODATA` to the variable `A_is_nodata`
+                // convert `A IS NODATA` to the check for `A.is_none()`
 
                 let mut pairs = pair.into_inner();
 
                 let identifier = pairs
                     .next()
                     .ok_or(ExpressionError::MissingIdentifier)?
-                    .as_str();
+                    .as_str()
+                    .into();
 
-                let new_identifier = format!("{}_is_nodata", identifier).into();
-
-                if self.boolean_parameters.contains(&new_identifier) {
-                    Ok(BooleanExpression::Variable(new_identifier))
-                } else {
-                    Err(ExpressionError::UnknownBooleanVariable {
-                        variable: identifier.to_string(),
-                    })
-                }
+                Ok(BooleanExpression::Comparison {
```

Comment:
does this mean `A = NODATA` is also valid in our DSL?

Reply:
yes

Reply:
that's what I meant. Is this desired behavior? In SQL you can't do `= NULL`, only `IS NULL`

Reply:
We should not tell that == works, but in SQL that always leads to confusion.